CMake/ExternalProjectGenerateProjectDescription.cmake

```diff
@@ -59,33 +72,97 @@ function(ExternalProject_GenerateProjectDescription_Step projectname)
     set(name ${_epgpd_NAME})
   endif()
 
-  set(description_file "version-${name}.txt")
+  set(generated_version_file "version-${name}.txt")
+  set(generated_license_file "license-${name}.txt")
+
+  set(extract_git_version 1)
 
   if(_epgpd_VERSION)
-    file(WRITE ${CMAKE_BINARY_DIR}/${description_file} "${name} ${_epgpd_VERSION}")
-    return()
+    file(WRITE ${CMAKE_BINARY_DIR}/${generated_version_file} "${name} ${_epgpd_VERSION}")
+    set(extract_git_version 0)
   endif()
 
-  set(script "${CMAKE_BINARY_DIR}/CMakeFiles/${projectname}-generate-project-description.cmake")
+  set(multiple_licenses 1)
 
-  file(WRITE "${script}"
-"execute_process(
-  COMMAND \"${GIT_EXECUTABLE}\" describe --always
-  OUTPUT_VARIABLE output
-  )
-file(WRITE \"${CMAKE_BINARY_DIR}/${description_file}\" \"${name} \${output}\")
-")
+  if(NOT _epgpd_LICENSE_FILES)
+    foreach(filename IN ITEMS NOTICE COPYRIGHT Copyright COPYING LICENSE License)
```

Comment:
I think instead of this hack, which does not guarantee presence of the actual license in the content, we introduce a requirement that 

1) every extension has a LICENSE file that should contain the license text (note that [current extension contribution instructions](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/BuildTestPackageDistributeExtensions) don't even mention "license"!)

2) presence of the LICENSE in the source code repository is a mandatory, and extension should not be merged until that file is present and content is valid

Reply:
This PR doesn't address the case of Slicer extension but focus on external project (e.g OpenSSL, ...). The  build will fail if no license information was found.

That said, here are some answer to your questions:

> every extension has a LICENSE file that should contain the license text (note that current 
> extension contribution instructions don't even mention "license"!)

While this is not explicitly listed in the step, the reviewer integrating extension to the index should check for this.

To find out about the requirement, clicking on the link at the top of the contribution page _To learn what is an extension, see [here](https://www.slicer.org/wiki/Documentation/Nightly/Developers/FAQ#What_is_an_extension_.3F)_ will outline the different level of compliance for extension. 

Adding a checklist for reviewer would be a great idea. Would  you like to take the lead on this ? 


> presence of the LICENSE in the source code repository is a mandatory, and extension should 
> not be merged until that file is present and content is valid

Agreed

Reply:
Makes sense, I missed that part. Sure, you can assign the issue to me. Working on a grant deadline today.

Reply:
Nice work!

I'm not sure the heuristics is needed. We could just skip license information if no LICENSE_FILES is specified.

Is there an option to specify custom string or local repository file (not in the external project's repository), just in case none of the files in the external repository are usable? (license is provided in a portal word file, license is not in separate file, etc.)